zellij-server/src/screen.rs

```diff
@@ -965,11 +966,11 @@ impl Screen {
         self.render()
     }
 
-    fn unblock_input(&self) {
+    fn unblock_input(&self) -> Result<()> {
         self.bus
             .senders
             .send_to_server(ServerInstruction::UnblockInputThread)
-            .unwrap();
+            .context("failed to send message to server")
```

Comment:
It doesn't seem like `send_to_server` (and other similar functions) are attaching that kind of context to the error. In order to support that we'd need to change the return type for those but it'll definitely break patterns like these https://github.com/zellij-org/zellij/blob/716f606b44f5014ef92b3896b0849143cd237891/zellij-server/src/screen.rs#L835-L840

I'll probably try to see how many places with this type of usage you have

Reply:
ok, it's just one place where it's used that way. Interestingly, I had to move this piece of code into `send_to_pty` because of missing traits for it's instruction. Not sure if it makes it better but I guess it's easier for you to take a look at the implementation and for me to revert that particular commit if it doesn't work that to operate on theory

Reply:
@arriven Yeah, that's a funny story. It stems from the fact that the `PluginInstruction` we send there is an enum, where one of its variants holds a `mpsc::Sender`. `anyhow`s `.context` & friends requires the error types they get to be `'static + Send + Sync` (And in our case the error type is `SendError<(PluginInstruction, _)>`), but [`mpsc::Sender` is `!Sync`][1]. That's why I made use of this "crutch" instead and construct an `anyhow::Error` from scratch. It's not pretty to look at, but it needn't be imo.

[1]: https://doc.rust-lang.org/stable/std/sync/mpsc/struct.Sender.html